src/traits.rs

```diff
@@ -233,6 +233,33 @@ pub trait TagExt: Accessor + Into<Tag> + Sized {
 	fn clear(&mut self);
 }
 
+/// Split and rejoin tags.
+///
+/// Useful and required for implementing read/modify/write round trips.
+pub trait SplitAndRejoinTag {
+	/// Extract and split generic contents into a [`Tag`].
+	///
+	/// Leaves the remainder that cannot be represented in the
+	/// resulting tag in `self`. This is useful if the modified [`Tag`]
+	/// is rejoined later using [`SplitAndRejoinTag::rejoin_tag`].
+	///
+	/// This operation leaves a partial tag behind that should only be
+	/// used again after rejoining a tag. Using the
+	/// [typestate pattern](http://cliffle.com/blog/rust-typestate/)
+	/// to represent the intermediate state turned out as less flexible
+	/// and useful than expected.
```

Comment:
```suggestion
	// NOTE: Using the "typestate pattern" (http://cliffle.com/blog/rust-typestate/)
	// to represent the intermediate state turned out as less flexible
	// and useful than expected.
```
This is only really useful to anyone trying to implement this in the future, it should just be a normal comment.